### PR TITLE
[ENH] HampelFilter detector for anomaly detection

### PR DESCRIPTION
**Pull request by Vbhatt03**
#### Reference Issues/PRs:

Adds HampelFilter, as mentioned in #6481 

#### What does this implement/fix? Explain your changes.

This PR implements the **Hampel Filter** for robust anomaly detection in time series data.

**Key Features:**
- Statistical outlier detection using rolling median and MAD (Median Absolute Deviation)
- Two algorithm variants:
  - **Standard Hampel**: Per-window MAD computation (O(N·w·log w))
  - **mMAD variant**: Efficient approximation using rolling median on deviations (O(N·w))
- Flexible windowing: Symmetric (centered) or trailing (causal) windows
- Configurable sensitivity via `n_sigmas` threshold parameter

**Implementation Details:**
- Follows sktime's `BaseDetector` architecture and conventions
- Task: `anomaly_detection`, Learning type: `unsupervised`
- Returns sparse output (anomaly indices) via `predict()`, dense output (all labels) via `transform()`
- Properly handles univariate Series and single-column DataFrames
- Includes parameter validation and automatic window size adjustment for centered windows

**Reference:** Implementation follows the approach described in [this paper](https://www.mdpi.com/1424-8220/25/11/3319).

#### Does your contribution introduce a new dependency? If yes, which one?

**No** - Uses only core dependencies (pandas, numpy) already in sktime.

#### What should a reviewer concentrate their feedback on?

- **Algorithm correctness**: Verify the MAD calculation and outlier detection logic
- **Code style**: Ensure compliance with sktime patterns and conventions
- **Test coverage**: Review the 23 test cases for comprehensiveness
- **Documentation**: Check docstrings for clarity and completeness
- **Parameter handling**: Validate window size adjustments and edge case handling
- **mMAD variant**: Confirm the computational efficiency improvement is correctly implemented

#### Did you add any tests for the change?

**Yes** - Added comprehensive test suite (`test_hampel.py`) with 23 tests covering:

**Basic functionality** (2 tests)
- Known outlier detection with different data patterns
- Input mutation check (data integrity)

**Edge cases** (3 tests)
- Flat signals (no outliers)
- Short series (< window_size)
- Data without anomalies

**Parameter variations** (8 tests)
- Multiple window sizes (3, 5, 7, 11)
- Different thresholds (n_sigmas)
- Standard vs mMAD variants
- Parameter combinations

**Algorithm features** (5 tests)
- Centered vs trailing windows
- Multiple outlier detection
- Threshold sensitivity
- Single-column DataFrames

**Output formats** (3 tests)
- Dense output (`transform`)
- Sparse output (`predict`)
- Format consistency

**Test framework** (1 test)
- `get_test_params()` validation

All tests pass locally.

#### Any other comments?

This is part of my preparation for an ESOC/GSoC proposal to expand sktime's detection module. I plan to implement additional detectors (Isolation Forest, Wild Binary Segmentation) and would appreciate feedback on the implementation approach before proceeding further.

**Next steps if this is merged:**
1. Add HampelFilter to API documentation (`docs/source/api_reference/detection.rst`)
2. Add illustrative usage examples to docstring
3. Consider multivariate extension for future work

I'm particularly interested in feedback on:
- Whether the mMAD variant adds sufficient value to justify the additional parameter
- Suggestions for improving the docstring examples
- Any edge cases I might have missed in testing

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
- [ ] Optionally, for added estimators: I've added myself to the `maintainers` tag
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]
  - **Title**: `[ENH] Add HampelFilter detector for anomaly detection`

##### For new estimators
- [ ] I've added the estimator to the API reference - in `docs/source/api_reference/detection.rst`
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section
- [x] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag
  - N/A - uses only core dependencies

### CODE REVIEW

File: sktime/detection/hampel.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+        median_series = X.rolling(self.window_size, center=self.center).median()
+        abs_dev = (X - median_series).abs()
+
+        if self.use_mmad:
+            mmad_window = self.mmad_window or self.window_size
+            mad_series = abs_dev.rolling(mmad_window, center=self.center).median()
+        else:
+            mad_series = abs_dev.rolling(self.window_size, center=self.center).median()
+
```
[the remark this answers is not in the document]
Comment (Vbhatt03) [reply sktime/detection/hampel.py:117]: The standard Hampel is already computed per window: rolling median on
abs deviations with the same window_size is equivalent to MAD per window.
mMAD differs only by using a secondary rolling median with a possibly different
window (mmad_window). Updated docstring to clarify this.

**Comment by Vbhatt03** [comment]
@fkiraly can you please re-run the CI tests, the patch should fix the issues

File: sktime/detection/hampel.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import numpy as np
+import pandas as pd
+
+from sktime.detection.base import BaseDetector
+
+__author__ = ["Vbhatt03"]
+__all__ = ["HampelAnomalyDetector"]
+
+
+class HampelAnomalyDetector(BaseDetector):
```
Comment (fkiraly) [inline sktime/detection/hampel.py]: what about `HampelDetector` to make it shorter?

**Comment by fkiraly** [comment]
failure is due to the docstring example

File: sktime/detection/hampel.py (hunk 2 above)
Comment (Vbhatt03) [reply sktime/detection/hampel.py]: sure, will change that and also fix the whitespace issue in docstring causing the failures

**Comment by Vbhatt03** [comment]
Done, @fkiraly. Kindly re-run the CI tests at your convenience.

**Comment by Vbhatt03** [comment]
Hi @fkiraly a gentle reminder for re-running the ci tests on this one. Thanks.

**Comment by Vbhatt03** [comment]
@fkiraly I have integrated the pr with current main. The CI triggered earlier by you didn't run due to the same. Could you please trigger the tests once more.
